tests/test_sequence_interpolator.py

```diff
@@ -0,0 +1,429 @@
+        )
+    ) as (timestamps, data, shift, seq_interp):
+        assert isinstance(
+            seq_interp, PhaseShiftedSequenceInterpolator
+        ), "Interpolation object is not a PhaseShiftedSequenceInterpolator"
+
+        delta_t = 1.0 / sampling_rate
+        times = (
+            timestamps[1:11] + 1e-9
+        )  # Add a small epsilon to avoid floating point errors
+        interp, valid = seq_interp.interpolate(times=times)
+        assert times.shape == valid.shape, "All samples should be valid"
+        assert np.allclose(
+            interp, data[0:10]
+        ), "Nearest neighbor interpolation does not match expected data"
+        assert valid.shape == (10,), f"Expected valid.shape == (10,), got {valid.shape}"
+        assert interp.shape == (
+            10,
+            n_signals,
+        ), f"Expected interp.shape == (10, {n_signals}), got {interp.shape}"
+
+        # Test phase shifts
+        for i in range(data.shape[1]):
+            for dt in np.linspace(0, 0.99) * delta_t:
+                shifted_times = times + shift[i] + dt
+
+                interp, valid = seq_interp.interpolate(times=shifted_times)
+                assert np.allclose(
+                    interp[:, i], data[1:11, i]
+                ), f"Data at {dt} does not match original data (use_mem_mapped={use_mem_mapped}, sampling_rate={sampling_rate}, shifts_per_signal={True})"
+
+            for dt in np.linspace(1.0, 1.99) * delta_t:
+                shifted_times = times + shift[i] + dt
+
+                interp, valid = seq_interp.interpolate(times=shifted_times)
+                assert np.allclose(
+                    interp[:, i], data[2:12, i]
+                ), f"Data at {dt} does not match original data (use_mem_mapped={use_mem_mapped}, sampling_rate={sampling_rate}, shifts_per_signal={True})"
+
+
+@pytest.mark.parametrize("n_signals", [0, 1, 10, 50])
+@pytest.mark.parametrize("keep_nans", [False, True])
+def test_nearest_neighbor_interpolation_with_phase_shifts_handles_nans(
+    n_signals, keep_nans
+):
+    with sequence_data_and_interpolator(
+        data_kwargs=dict(
+            n_signals=n_signals,
+            use_mem_mapped=True,
+            t_end=5.0,
+            sampling_rate=10.0,
+            shifts_per_signal=True,
+            contain_nans=True,
+        ),
+        interp_kwargs=dict(keep_nans=keep_nans),
+    ) as (timestamps, data, _, seq_interp):
+        assert isinstance(
+            seq_interp, PhaseShiftedSequenceInterpolator
+        ), "Interpolation object is not a PhaseShiftedSequenceInterpolator"
+
+        times = (
+            timestamps[1:11] + 1e-9
+        )  # Add a small epsilon to avoid floating point errors
+        interp, valid = seq_interp.interpolate(times=times)
+        assert times.shape == valid.shape, "All samples should be valid"
+        assert np.allclose(
+            interp, data[0:10], equal_nan=True
+        ), "Nearest neighbor interpolation does not match expected data"
```

Comment:
exact here instead of `allclose` (but just == would not work here because of nans, I guess)